cmd/infra/aws/iam.go

```diff
@@ -320,109 +318,66 @@ func (o *CreateIAMOptions) CreateOIDCResources(iamClient iamiface.IAMAPI, s3Clie
 		Bytes:   x509.MarshalPKCS1PrivateKey(privKey),
 	})
 
-	pubKey := &privKey.PublicKey
-	pubKeyDERBytes, err := x509.MarshalPKIXPublicKey(pubKey)
+	issuerURLWithProto := o.IssuerURL
+	url, err := url.Parse(issuerURLWithProto)
 	if err != nil {
+		log.Error(err, "Failed to parse issuer-url", "issuer-url", issuerURLWithProto)
```

Comment:
This is both fatal and can be computed during very early flag validation, so could it be moved way up to validation before we even attempt to do any work?

Reply:
This is already done very early.  Only thing done before this is signing key generation.

Reply:
What I mean is we should never even get this far, it's a static validation that should have happened immediately after flag parsing and before `Run()` was ever invoked. I agree it happens that there's no side effect today before this current point. Probably should adopt a Validate() type convention in these commands generally for this stuff

Reply:
(for future ref, I wonder if there's some way to plug in a URL flag type to cobra)

Reply:
I tired to implement this as a `PreRunE` on the `cobra.Command` but this call from `create cluster` circumvents it
https://github.com/openshift/hypershift/blob/main/cmd/cluster/create.go#L214